### ISSUE TITLE
Add the ability to warn against usage of certain tagged literals

### ISSUE THREAD

**Issue by GlassAndOneHalf** [clj-kondo/clj-kondo#2030]
To upvote this issue, give it a thumbs up. See [this list](https://github.com/clj-kondo/clj-kondo/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc) for the most upvoted issues.

**Is your feature request related to a problem? Please describe.**

I have been converting a codebase to use the new Java time APIs, replacing `java.util.Date`. To discourage developers from using `java.util.Date` in the future, I would like to be able to warn against any usage of the `#inst` tagged literal, as this constructs `java.util.Date` instances.

**Describe the solution you'd like**

Ideally a linter that allows me to specify a tagged literal that should be flagged as a warning/error whenever it is used.

For example, in the `clj-kondo` config:
```
:linters {
  :discouraged-tag { inst { :message "Prefer #java-time/instant over #inst" } }
}
```

**Describe alternatives you've considered**
N/A

**Additional context**
N/A
